src/parser/mod.rs

```diff
@@ -18,3 +20,65 @@ pub enum ExtendedTerm {
     RichTerm(RichTerm),
     ToplevelLet(Ident, RichTerm),
 }
+
+impl grammar::ExtendedTermParser {
+    pub fn parse_term_tolerant(
+        &self,
+        file_id: FileId,
+        lexer: lexer::Lexer,
+    ) -> Result<(ExtendedTerm, ParseErrors), ParseError> {
+        let mut parse_errors = Vec::new();
+        let result = self
+            .parse(file_id, &mut parse_errors, lexer)
+            .map_err(|err| ParseError::from_lalrpop(err, file_id));
+
+        let parse_errors = ParseErrors::from_recoverable(parse_errors, file_id);
+        match result {
+            Ok(t) => Ok((t, parse_errors)),
+            Err(e) => Err(e),
+        }
+    }
+
+    pub fn parse_term(
```

Comment:
Nitpick of nitpicks: could be named just `parse` and `parse_term`, as the name of the structure `TermParser` is already quite explicit.

Reply:
Unfortunately, both structs already have a lalrpop generated method `parse`.
I'm also not happy with the naming scheme, but couldn't think of anything else?

Reply:
I see. Then we can either leave it as it is, or wrap LALRPOP "raw" parsers in new structs, so that we can hide the default `parse` method and only exposes the current `parse_term` and `parse_term_tolerant`. But then, you have to find a good name for the wrapping structure :stuck_out_tongue: